internal/dependencyinjection/admin_db_connection_pool.go

```diff
@@ -0,0 +1,40 @@
+package dependencyinjection
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stellar/go/support/log"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/db"
+	"github.com/stellar/stellar-disbursement-platform-backend/db/router"
+)
+
+const AdminDBConnectionPoolInstanceName = "admin_db_connection_pool_instance"
+
+// NewAdminDBConnectionPool creates a new admin db connection pool instance, or retrives an instance that was already
+// created before.
+func NewAdminDBConnectionPool(ctx context.Context, opts DBConnectionPoolOptions) (db.DBConnectionPool, error) {
+	// If there is already an instance of the service, we return the same instance
+	instanceName := AdminDBConnectionPoolInstanceName
+	if instance, ok := GetInstance(instanceName); ok {
+		if dbConnectionPoolInstance, ok := instance.(db.DBConnectionPool); ok {
```

Comment:
I believe ok is shadowed here as well.

Reply:
Fixed in https://github.com/stellar/stellar-disbursement-platform-backend/pull/207/commits/361dcb74fad43f1e52d05a80091dd151be86e9e0.